modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/application/ApplicationBlockSubscriptionTestCase.java

```diff
@@ -0,0 +1,109 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.am.integration.tests.application;
+
+import org.junit.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.publisher.api.v1.SubscriptionsApi;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationDTO;
+import org.wso2.am.integration.test.impl.RestAPIStoreImpl;
+import org.wso2.am.integration.test.utils.UserManagementUtils;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.am.integration.tests.api.lifecycle.APIManagerLifecycleBaseTest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import java.net.URL;
+
+import static org.wso2.am.integration.test.utils.base.APIMIntegrationConstants.SUPER_TENANT_DOMAIN;
+
+public class ApplicationBlockSubscriptionTestCase extends APIManagerLifecycleBaseTest {
+    private static final String APPLICATION_NAME = "test-app";
+    private static final String USER_NAME = "test-user";
+    private static final String PASSWORD = "password";
+    private static final String ORGANIZATION = "test";
+    private static final String FIRST_NAME = "John";
+    private RestAPIStoreImpl restAPIStoreClientUser;
+    private String applicationId;
+    private String apiId;
+    private final String API_END_POINT_POSTFIX_URL = "jaxrs_basic/services/customers/customerservice/";
+    private SubscriptionsApi subscriptionsApi;
+    private String subscriptionId;
+
+    private static final String PROD_ONLY_BLOCK_STATE = "PROD_ONLY_BLOCKED";
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public ApplicationBlockSubscriptionTestCase(TestUserMode userMode) {
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init(userMode);
+        subscriptionsApi = new SubscriptionsApi(restAPIPublisher.apiPublisherClient);
+        createUsersAndApplications();
+    }
+
+    @Test(description = "To test block functionality works for the application name and owner name which contains hyphen")
+    public void testBlockUnblockSubscription() throws Exception {
+        int statusCodeBlock = subscriptionsApi.blockSubscriptionWithHttpInfo(subscriptionId,
+                PROD_ONLY_BLOCK_STATE, null).getStatusCode();
```

Comment:
Better to add new assert for  Api invocation with subscription blocking and unblocking?

Reply:
Resolving with this PR: https://github.com/wso2/product-apim/pull/13025